### DISCUSSION TITLE
Failing to run tests. Can someone help me with it?

### DISCUSSION THREAD

**Discussion by minitesh** [js-fns/date-fns#2668] [Q&A]
I just now forked repo and installed dependencies which got succeeded.
Now I am trying to run tests then its giving me this error

λ yarn test
yarn run v1.21.1
$ karma start config/karma.js
06 10 2021 23:16:38.153:ERROR [config]: Error in config file!
  Error: Cannot find module 'webpack'
Require stack:
- C:\Personal\date-fns\config\webpack.js
- C:\Personal\date-fns\config\karma.js
- C:\Users\nitesh_phadatare\AppData\Local\Yarn\Data\global\node_modules\karma\lib\config.js
- C:\Users\nitesh_phadatare\AppData\Local\Yarn\Data\global\node_modules\karma\lib\server.js
- C:\Users\nitesh_phadatare\AppData\Local\Yarn\Data\global\node_modules\karma\lib\cli.js
- C:\Users\nitesh_phadatare\AppData\Local\Yarn\Data\global\node_modules\karma\bin\karma
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:966:15)
    at Function.Module._load (internal/modules/cjs/loader.js:842:27)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (C:\Personal\date-fns\config\webpack.js:2:17)
    at Module._compile (internal/modules/cjs/loader.js:1138:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
    at Module.load (internal/modules/cjs/loader.js:986:32)
    at Function.Module._load (internal/modules/cjs/loader.js:879:14)
    at Module.require (internal/modules/cjs/loader.js:1026:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by fturmel** [comment]
Yarn v1.21.1 is pretty old (2+ years), maybe it'll help to upgrade your yarn and node versions to latest LTS?

Then start again fresh by deleting the project's node_modules directory, run `yarn cache clean` and then `yarn install` and see if that works.

If you still can't make it work, using Jest should be OK. Try `yarn jest` instead of `yarn test`. Both Karma (chromium) and Jest (node) runners are used on the CI pipeline to execute all tests.

**Reply by minitesh** [reply]
I have updated yarn version to v1.22.15 but still getting the same error

**Reply by fturmel** [reply]
Looking more carefully at your error stack, it seems like it's executing a globally installed version of karma.

I was able to recreate your error by doing the following:
- delete the project's node_modules directory (removing all dependencies)
- install karma globally only: `yarn global add karma`
- run `yarn test` inside the project

To me, that implies that you didn't actually install the project's dependencies to begin with, or maybe yarn had some issue linking the karma binary during install and incorrectly uses the global version when you run `yarn test`.

Have you tried to remove that global version, or look more carefully for errors during the `yarn install` process?

Here's what a fresh setup looks like on my machine, maybe that will help:

`git clone git@github.com:date-fns/date-fns.git && cd date-fns && yarn install && yarn test`

<img width="903" alt="Screen Shot 2021-10-21 at 12 49 44 PM" src="https://user-images.githubusercontent.com/4009209/138322184-d48c0df7-ecc1-4ffe-aa57-2ecaaa056ecf.png">
